folio-org/folio-api-tests
# MODKBEKBJ-54: Add mod kb ebsco java tests

Copied tests for module mod-kb-ebsco-java from mod-kb-ebsco
Adjusted tests to pass on mod-kb-ebsco-java
Made following changes to tests:
1) removed providerToken check, because providerToken is not returned for
most of providers
2) changed expected error messages
3) removed publisherName check for titles, because it is usually not present too
4) changed test for invalid provider id ( for path "/eholdings/providers/and" )
to verify that response has status code 400 instead of 500
5) changed some tests to expect error message in plain text instead of in the form of json, some invalid parameters can't be parsed properly so in java module they result in immediate error, we can't control this logic so error message can only be returned in plain text.

Compared to mod-kb-ebsco following tests are removed for now, because corresponding endpoints are not implemented:
1) setup for resources test
2) resources
3) tear-down resources test
3) tear-down titles test/Delete Custom Package
4) setup for titles test/Create Custom Package
5) setup for titles test/GET Sample Managed Package
6) providers/PUT provider by providerId
7) providers/GET provider by providerId including packages
8) titles/GET title by titleId
9) titles/POST title by titleId
10) providers / GET provider by providerId / Positive / with valid providerId including packages

Result of running tests locally:
```
┌─────────────────────────┬──────────┬──────────┐
│                         │ executed │   failed │
├─────────────────────────┼──────────┼──────────┤
│              iterations │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│                requests │       40 │        0 │
├─────────────────────────┼──────────┼──────────┤
│            test-scripts │       87 │        1 │
├─────────────────────────┼──────────┼──────────┤
│      prerequest-scripts │       55 │        0 │
├─────────────────────────┼──────────┼──────────┤
│              assertions │      225 │        0 │
├─────────────────────────┴──────────┴──────────┤
│ total run duration: 31.3s                     │
├───────────────────────────────────────────────┤
│ total data received: 162.88KB (approx)        │
├───────────────────────────────────────────────┤
│ average response time: 728ms                  │
└───────────────────────────────────────────────┘

  #  failure    detail                                                                      
                                                                                            
 1.  TypeError                                                                              
                Cannot read property 'toLowerCase' of undefined                             
                at test-script                                                              
                inside "titles / GET title collection / Positive / /titles filter[subject]" 
```
